File: tests/Aiko.Core.Tests/AikoFoldersTests.cs
using Aiko.Core;

namespace Aiko.Core.Tests;

/// The folders the app, the bridge and the shim share. Every path here is on people's disks
/// already, so each one is written out in full: a changed name would lose their settings or split
/// the bridge from the tray.
public class AikoFoldersTests
{
    private static readonly AikoFolders Folders = AikoFolders.Windows(
        @"C:\Users\someone\AppData\Roaming", @"C:\Users\someone\AppData\Local");

    [Fact]
    public void The_persons_choices_live_under_APPDATA_Aiko()
    {
        Assert.Equal(@"C:\Users\someone\AppData\Roaming\Aiko", Folders.SettingsFolder);
        Assert.Equal(@"C:\Users\someone\AppData\Roaming\Aiko\settings.json", Folders.SettingsFile);
        Assert.Equal(@"C:\Users\someone\AppData\Roaming\Aiko\environments.json", Folders.EnvironmentsFile);
        Assert.Equal(@"C:\Users\someone\AppData\Roaming\Aiko\persona.json", Folders.PersonaFile);
        Assert.Equal(@"C:\Users\someone\AppData\Roaming\Aiko\install-id", Folders.InstallIdFile);
        Assert.Equal(@"C:\Users\someone\AppData\Roaming\Aiko\reported.json", Folders.ReportedPeriodsFile);
    }

    [Fact]
    public void What_Aiko_makes_for_itself_lives_under_LOCALAPPDATA_Aiko()
    {
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko", Folders.LocalFolder);
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko\environments", Folders.SnapshotsFolder);
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko\activity", Folders.ActivityFolder);
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko\direct", Folders.DirectCacheFolder);
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko\bin", Folders.CommandsFolder);
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko\marketplace", Folders.MarketplaceFolder);
        Assert.Equal(
            @"C:\Users\someone\AppData\Local\Aiko\marketplace\.claude-plugin\marketplace.json",
            Folders.MarketplaceFile);
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko\plugins", Folders.PersonaPluginsFolder);
        Assert.Equal(@"C:\Users\someone\AppData\Local\Aiko\log.txt", Folders.LogFile);
    }

    [Fact]
    public void The_installed_app_is_where_Velopack_puts_it()
    {
        Assert.Equal(@"C:\Users\someone\AppData\Local\Slayumind.Aiko\current", Folders.InstalledAppFolder);
    }

    [Fact]
    public void A_snapshot_file_is_named_after_the_snapshot()
    {
        Assert.Equal(
            @"C:\Users\someone\AppData\Local\Aiko\environments\claude-work.json",
            Folders.SnapshotFile(SnapshotName.For(@"C:\Users\someone\.claude-work\")));
    }

    [Fact]
    public void A_second_environment_folder_sits_beside_claude()
    {
        Assert.Equal(".claude-work", ClaudeConfigFolder.NamedFolderName("work"));
        Assert.Equal(".claude*", ClaudeConfigFolder.SearchPattern);
    }
}

File: mac/Tests/AikoKitTests/AikoFoldersTests.swift
import Testing

@testable import AikoKit

/// The folders the app, the bridge and the shim share. Every path here is on people's disks
/// already, so each one is written out in full: a changed name would lose their settings or split
/// the bridge from the tray.
struct AikoFoldersTests {
    static let folders = AikoFolders.windows(
        #"C:\Users\someone\AppData\Roaming"#, #"C:\Users\someone\AppData\Local"#)

    @Test
    func thePersonsChoicesLiveUnderAppDataAiko() {
        let roaming = #"C:\Users\someone\AppData\Roaming\Aiko"#

        #expect(Self.folders.settingsFolder == roaming)
        #expect(Self.folders.settingsFile == roaming + #"\settings.json"#)
        #expect(Self.folders.environmentsFile == roaming + #"\environments.json"#)
        #expect(Self.folders.personaFile == roaming + #"\persona.json"#)
        #expect(Self.folders.installIdFile == roaming + #"\install-id"#)
        #expect(Self.folders.reportedPeriodsFile == roaming + #"\reported.json"#)
    }

    @Test
    func whatAikoMakesForItselfLivesUnderLocalAppDataAiko() {
        let local = #"C:\Users\someone\AppData\Local\Aiko"#

        #expect(Self.folders.localFolder == local)
        #expect(Self.folders.snapshotsFolder == local + #"\environments"#)
        #expect(Self.folders.activityFolder == local + #"\activity"#)
        #expect(Self.folders.directCacheFolder == local + #"\direct"#)
        #expect(Self.folders.commandsFolder == local + #"\bin"#)
        #expect(Self.folders.marketplaceFolder == local + #"\marketplace"#)
        #expect(Self.folders.marketplaceFile == local + #"\marketplace\.claude-plugin\marketplace.json"#)
        #expect(Self.folders.personaPluginsFolder == local + #"\plugins"#)
        #expect(Self.folders.logFile == local + #"\log.txt"#)
    }

    @Test
    func theInstalledAppIsWhereVelopackPutsIt() {
        #expect(Self.folders.installedAppFolder == #"C:\Users\someone\AppData\Local\Slayumind.Aiko\current"#)
    }

    @Test
    func aSnapshotFileIsNamedAfterTheSnapshot() {
        #expect(
            Self.folders.snapshotFile(SnapshotName.forConfigDirectory(#"C:\Users\someone\.claude-work\"#))
                == #"C:\Users\someone\AppData\Local\Aiko\environments\claude-work.json"#)
    }

    @Test
    func aSecondEnvironmentFolderSitsBesideClaude() {
        #expect(ClaudeConfigFolder.namedFolderName("work") == ".claude-work")
        #expect(ClaudeConfigFolder.searchPattern == ".claude*")
    }
}
